pkg/controller/externalnode/controller.go

```diff
@@ -300,7 +300,17 @@ func (c *ExternalNodeController) updateExternalNode(preEn *v1alpha1.ExternalNode
 func (c *ExternalNodeController) updateExternalEntity(ee *v1alpha2.ExternalEntity) error {
 	// resourceVersion must be specified for update operation,
 	// so it gets the existing ExternalEntity and modifies the changed fields.
-	existingEE, _ := c.crdClient.CrdV1alpha2().ExternalEntities(ee.Namespace).Get(context.TODO(), ee.Name, metav1.GetOptions{})
+	existingEE, err := c.crdClient.CrdV1alpha2().ExternalEntities(ee.Namespace).Get(context.TODO(), ee.Name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			_, err = c.crdClient.CrdV1alpha2().ExternalEntities(ee.Namespace).Create(context.TODO(), ee, metav1.CreateOptions{})
+			if err == nil {
+				return nil
+			}
+		}
+		klog.ErrorS(err, "Failed to update ExternalEntity", "entityName", ee.Name, "entityNameSpace", ee.Namespace)
```

Comment:
The log could be confusing when either Get or Create fails as it says "Failed to update".

Reply:
updated.